google/cloud/forseti/services/scanner/dao.py

```diff
@@ -190,3 +201,40 @@ def map_by_resource(violation_rows):
             v_by_type[v_resource].append(v_data)
 
     return dict(v_by_type)
+
+def _create_violation_hash(violation_full_name, inventory_data, violation_data):
+    """Create a hash of violation data using the requested algorithm.
+
+    Args:
+        violation_full_name (str): The full name of the violation.
+        inventory_data (str): The inventory data.
+        violation_data (dict): A violation.
+
+    Returns:
+        str: The resulting hex digest or '' if we can't successfully create
+        a hash.
+    """
+
+    # TODO: Intelligently choose from hashlib.algorithms_guaranteed if our
+    # desired one is not available.
+    algorithm = 'sha512'
+
+    try:
+        violation_hash = hashlib.new(algorithm)
+    except ValueError as e:
```

Comment:
Do we still need this try/except if we use a default ```sha512``` that should just work?

Reply:
It's not _guaranteed__ to work on all systems as noted in the comment on :L218 this can be improved to parse that output of algorithms_guaranteed and choose intelligently from it.